business/health.go

```diff
@@ -291,9 +293,23 @@ func fillWorkloadRequestRates(allHealth models.NamespaceWorkloadHealth, rates mo
 func (in *HealthService) getServiceRequestsHealth(namespace, service, rateInterval string, queryTime time.Time) (models.RequestHealth, error) {
 	rqHealth := models.NewEmptyRequestHealth()
 	inbound, err := in.prom.GetServiceRequestRates(namespace, service, rateInterval, queryTime)
+	if err != nil {
+		return rqHealth, err
+	}
 	for _, sample := range inbound {
 		rqHealth.AggregateInbound(sample)
 	}
+
+	svc, err := in.k8s.GetService(namespace, service)
```

Comment:
Ok, there is a missing case in the kialiCached that I missed.
This query is ok, but it should be done via cache.
No changes for the moment as this requires more changes.

I'll send a followup PR https://github.com/kiali/kiali/issues/3467 and I will update this later.

Reply:
Note, that PR #3467 is merged so business package can get this from cached.
i.e. used from validations
```
svc, err := in.businessLayer.Svc.getService(namespace, service)
```